src/dipdup/context.py

```diff
@@ -35,11 +40,36 @@ def reset(self) -> None:
     def updated(self) -> bool:
         return self._updated
 
+    async def restart(self) -> None:
+        """Restart preserving CLI arguments"""
+        # NOTE: Remove --reindex from arguments to avoid reindexing loop
+        if '--reindex' in sys.argv:
+            sys.argv.remove('--reindex')
+        os.execl(sys.executable, sys.executable, *sys.argv)
+
     async def reindex(self) -> None:
```

Comment:
Maybe it would be more explicit/imperative to specify tables that should be dropped/preserved in reindex?
E.g. 
```
def reindex(preserve_tables=None)
```